deltacat/compute/stats/utils/intervals.py

```diff
@@ -1,62 +1,88 @@
-from typing import Set, Tuple
-from collections import Iterable
+from typing import Set, Tuple, List, Iterable, Union
 
+UnboundedRange = Union[int, None]
+NumericalRange = Union[int, float]
 
-# TODO: Allow users to add infinite ranges in intervals: i.e. (5, None) => (5, math.inf)
-def merge_intervals(intervals: Set[Tuple[int, int]]):
+
+def merge_intervals(intervals: Set[Tuple[UnboundedRange, UnboundedRange]]) -> \
+        Set[Tuple[UnboundedRange, UnboundedRange]]:
     """Merges a set of N input intervals into a minimal number of output intervals.
 
-    Each input interval must be closed bounded intervals. Intervals will be merged into
-    a minimal number of closed bounded output intervals in O(N log N) time.
+    Each input interval is an open unbounded interval with infinite ranges.
+    Intervals will be merged into a minimal number of output intervals in O(N log N) time.
 
     Example:
         >>> merge_intervals((3, 9), (8, 12), (15, 19))
         ((3, 12), (15, 19))
 
+    Example:
+        >>> merge_intervals((3, 9), (None, 15), (13, 30))
+        ((None, 30))
```

Comment:
Maybe add one more example showing that `math.inf` is also equivalent to None?

Reply:
See above comment - let's just drop `math.inf` support. Seems to overcomplicate the code without giving back much of any tangible benefits.

Reply:
Yeah I agree that from a client caller perspective, `math.inf` doesn't seem as intuitive to pass in, as opposed to `None` to represent all values for a start/end interval.